partner_default_sale_discount/models/sale_order.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    default_sale_discount = fields.Float(
+        related="partner_id.commercial_partner_id.default_sale_discount",
+        string="Default sales discount (%)",
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        """Inject the default in the context of the line this way for
+        making it inheritable.
+        """
+        res = super(SaleOrder, self).fields_view_get(
+            view_id=view_id, view_type=view_type, toolbar=toolbar,
+            submenu=submenu
+        )
+        if view_type != 'form':  # pragma: no cover
+            return res
+        eview = etree.fromstring(res['arch'])
+        xml_order_line = eview.xpath("//field[@name='order_line']")
+        xml_discount = eview.xpath("//field[@name='default_sale_discount']")
+        if xml_order_line and xml_discount:
+            # This should be handled in "string" mode
+            context = xml_order_line[0].get('context', '{}')
+            index = context.find('{')
+            context = (
+                context[:index + 1] +
+                "'default_discount': default_sale_discount, " +
+                context[index + 1:]
+            )
+            xml_order_line[0].set('context', context)
```

Comment:
Because context can contain JS keys that can't be interpreted with eval. Take for example this injected context as example. We have the variable `default_sale_discount`, that should be interpreted in "execution time" by the JS client, so if there are another variables like this one in the previous context, `safe_eval` will fail.